### PR TITLE
Don't transform whitelist remote locators [6543]

### PR DESCRIPTION
**Pull request by EduPonz**
don't transform remote locator to localhost if it's in the whitelist

### CODE REVIEW

File: src/cpp/fastrtps_deprecated/transport/TCPTransportInterface.cpp [hunk 1]
```diff
@@ -449,19 +449,27 @@ bool TCPTransportInterface::transform_remote_locator(
...
-        if (is_local_locator(remote_locator))
+        if (is_locator_allowed(remote_locator))
+        {
+            if (IPLocator::isLocal(remote_locator))
+            {
+                // Loopback locator
+                fill_local_ip(result_locator);
+                IPLocator::setPhysicalPort(result_locator, IPLocator::getPhysicalPort(remote_locator));
+                IPLocator::setLogicalPort(result_locator, IPLocator::getLogicalPort(remote_locator));
+                return true;
```
Comment (LuisGP) [inline src/cpp/fastrtps_deprecated/transport/TCPTransportInterface.cpp]: I miss the checking that `localhost` is allowed too:

```suggestion
                if (is_locator_allowed(result_locator))
                {
                    return true;
                }
```

Comment (LuisGP) [reply src/cpp/fastrtps_deprecated/transport/TCPTransportInterface.cpp]: This check should be added to the `else if` branch too.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/608/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/530/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/604/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/613/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/535/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/609/)

**Comment by raquelalvarezbanos** [comment]
Tests passed on all platforms (see [Windows](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/612/), [Linux](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/616/), and [Mac](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/538/)) although overall result is marked as failure.

**Comment by raquelalvarezbanos** [comment]
@EduPonz @laura-eprosima we should make a backward port against 1.9.x

**Pull request merged**